unlock-app/src/components/content/event/NewEvent.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react'
+import { AppLayout } from '~/components/interface/layouts/AppLayout'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useConfig } from '~/utils/withConfig'
+import { useWalletService } from '~/utils/withWalletService'
+import { Form, NewEventForm } from './Form'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { LockDeploying } from './LockDeploying'
+import { MetadataFormData } from '~/components/interface/locks/metadata/utils'
+
+export interface TransactionDetails {
+  hash: string
+  network: number
+}
+
+export const NewEvent = () => {
+  const { changeNetwork } = useAuth()
+  const walletService = useWalletService()
+  const config = useConfig()
+  const [transactionDetails, setTransactionDetails] =
+    useState<TransactionDetails>()
+  const [lockAddress, setLockAddress] = useState<string>()
+  const [metadata, setMetadata] = useState<MetadataFormData>()
+
+  const onSubmit = async (formData: NewEventForm) => {
+    // prompt the user to change network if applicable
+    await changeNetwork(formData.network)
+    setMetadata({
+      name: formData.lock.name,
+      ...formData.metadata,
+    })
+
+    let lockAddress
+    try {
+      lockAddress = await walletService.createLock(
+        {
```

Comment:
This does not work if you change network. 
Reproduction steps

1. Switch to optimism
2. Use goerli on event
3. Click on deploy lock

You will be prompted to switch but the next transaction will fail. 

The problem is you are still using walletService that is connected to the older network due to closures. 

Also the alternative doesn't where you directly use provider yields the network switch error I showed.

Reply:
what do you recommnend we do to fix it? reset the `walletService` inside of `changeNetwork(formData.network)` ?

Reply:
Copy what we do in the lock creation page. Ask user to switch when a new network is selected. I know it's not ideal but it's tested and reliable because this PR is already complex and need to launch today.

Reseting wallet service inside it wouldn't work as there will be another issue (I posted a link to the error related to that).

Reply:
OK let me do that then.

Reply:
fixed!